src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsStream.cs

```diff
@@ -66,50 +72,40 @@ public TdsStream(Stream underLyingStream, TdsWriteStream writeStream) : base()
         public void ReplaceUnderlyingStream(Stream stream)
         {
             _writeStream.ReplaceUnderlyingStream(stream);
-            // TODO: do this for the read stream as well.
+            _readStream.ReplaceUnderlyingStream(stream);
         }
 
         /// <summary>
         /// Allows manipulation of the packet size. This should only be used after login ack,
         /// when the client and server exchange the negotiated packet size.
         /// </summary>
         /// <param name="packetSize">The negotiated packet size</param>
-        /// <exception cref="NotImplementedException"></exception>
         public void SetPacketSize(int packetSize)
```

Comment:
Very tentative, but it's worth experimenting: when the packet size changes, perhaps the underlying socket's `ReceiveBufferSize` and `SendBufferSize` should change, accounting for TDS+MARS+SSL header sizes. I'm not sure whether this'd be an all-round performance improvement, or a CPU usage/network latency tradeoff.